lib/Command/MigrateCustomGroups.php

```diff
@@ -0,0 +1,256 @@
+use OCA\Circles\Exceptions\FederatedUserNotFoundException;
+use OCA\Circles\Exceptions\InvalidIdException;
+use OCA\Circles\Exceptions\MemberNotFoundException;
+use OCA\Circles\Exceptions\OwnerNotFoundException;
+use OCA\Circles\Exceptions\RemoteInstanceException;
+use OCA\Circles\Exceptions\RemoteNotFoundException;
+use OCA\Circles\Exceptions\RemoteResourceNotFoundException;
+use OCA\Circles\Exceptions\RequestBuilderException;
+use OCA\Circles\Exceptions\SingleCircleNotFoundException;
+use OCA\Circles\Exceptions\UnknownRemoteException;
+use OCA\Circles\Exceptions\UserTypeNotFoundException;
+use OCA\Circles\IFederatedUser;
+use OCA\Circles\Model\FederatedUser;
+use OCA\Circles\Model\Member;
+use OCP\DB\Exception;
+use OCP\DB\QueryBuilder\IQueryBuilder;
+use OCP\IConfig;
+use OCP\IDBConnection;
+use OCP\Share\IShare;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class MigrateCustomGroups extends Base {
+	private OutputInterface $output;
+	/** @var IFederatedUser[] */
+	private array $fedList = [];
+
+	public function __construct(
+		private CirclesManager $circlesManager,
+		protected IDBConnection $connection,
+		protected IConfig $config,
+		private LoggerInterface $logger,
+	) {
+		parent::__construct();
+	}
+
+	protected function configure() {
+		parent::configure();
+		$this->setName('circles:migrate:customgroups');
+	}
+
+	protected function execute(InputInterface $input, OutputInterface $output): int {
+		$this->output = $output;
+		if (!$this->shouldRun()) {
+			$this->output->writeln('migration already done or table \'custom_group\' not found');
+			return 0;
+		}
+
+		$this->migrateTeams();
+		$this->config->setAppValue('circles', 'imported_custom_groups', 'true');
+
+		return 0;
+	}
+
+	public function migrateTeams(): void {
+		$this->output->writeln('Migrating custom groups to Teams');
+
+		$owners = $this->extractCustomGroupsAndOwners();
+
+		// we get list of all custom groups
+		$queryCustomGroups = $this->connection->getQueryBuilder();
+		$queryCustomGroups->select('group_id', 'display_name', 'uri')
+						  ->from('custom_group')
+						  ->orderBy('group_id');
+
+		$resultCustomGroups = $queryCustomGroups->executeQuery();
+
+		// we cycle for each custom group
+		while ($rowCG = $resultCustomGroups->fetch()) {
+			$groupId = $rowCG['group_id'] ?? 0;
+			$groupUri = $rowCG['uri'] ?? '';
+			$ownerId = $owners[$groupId] ?? '';
+			if ($ownerId === '' || $groupId === 0) {
+				continue; // if group or owner is not know, we ignore the entry.
+			}
+
+			// based on owner's userid, we create federateduser and a new circle
+			$this->output->writeln('+ New Team <info>' . $rowCG['display_name'] . '</info>, owner by <info>' . $ownerId . '</info>');
+			$owner = $this->cachedFed($ownerId);
+
+			$this->circlesManager->startSession($owner);
+			$circle = $this->circlesManager->createCircle($rowCG['display_name']);
```

Comment:
This at least can throw `CircleNameTooShortException`, but probably it also throws if the name is already taken, no? We probably need some exception handling here.

Reply:
the too short exception might occurs, but in case of name taken a digit will be added

Reply:
(added an exception management)
also details about original data in the top comment